src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+
+import {DelegateRegistry} from './interfaces/IDelegateRegistry.sol';
+import {IVeToken} from './interfaces/IVeToken.sol';
+import {IWardenBoost} from './interfaces/IWardenBoost.sol';
+import {Core} from './Core.sol';
+
+abstract contract VeTokenManager is Core {
+  event BuyBoost(address delegator, address receiver, uint256 amount, uint256 duration);
+  event DelegateUpdate(address indexed oldDelegate, address indexed newDelegate);
+  event Lock(uint256 cummulativeTokensLocked, uint256 lockHorizon);
+  event Unlock(uint256 tokensUnlocked);
+
+  DelegateRegistry public constant DELEGATE_REGISTRY =
+    DelegateRegistry(0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446);
+
+  address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+  address public constant B_80BAL_20WETH = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+  address public constant VE_BAL = 0xC128a9954e6c874eA3d62ce62B468bA073093F25;
+  address public constant WARDEN_VE_BAL = 0x42227bc7D65511a357c43993883c7cef53B25de9;
+
+  /// @notice the snapshot delegate for the deposit
+  address public delegate;
+  /// @notice the keccak encoded spaceId of the snapshot space
+  bytes32 public spaceId;
+  /// @notice The lock duration of veToken
+  uint256 lockDuration;
+
+  /// @notice The Buyer will then need to call the user_checkpoint() method on all Curve V4 & Factory Gauges
+  /// currently farming, to apply the newly received Boost.
+  /// @notice duration in weeks (ie: 1 for 1 Week)
+  function buyBoost(
```

Comment:
Missing NatSpec, would also recommend replacing `duration` with `durationWeeks` or something similar to specify the unit in-parameter.

Reply:
Done.